lib/std/json.zig

```diff
@@ -1201,6 +1201,36 @@ pub const Value = union(enum) {
         w.one_indent = one_indent;
         try w.emitJson(self);
     }
+
+    /// json.Value equality comparison: for `.Array`s and `.Object`s, equal
+    /// sizes are prerequisite before further probing into their contents.
+    pub fn eql(self: Value, other: Value) bool {
+        if (std.meta.activeTag(self) == std.meta.activeTag(other)) switch (self) {
+            .Null => return true,
+            .Bool => |self_bool| return self_bool == other.Bool,
+            .Integer => |self_int| return self_int == other.Integer,
+            .Float => |self_float| return self_float == other.Float, // TODO: reconsider if/when std.math gets a dedicated float eql
```

Comment:
This shouldn't need a `float.eql`, but you should consider your desired behaviour on NaN

Reply:
On point. Now personally I don't have any "desired" behavior on either NaN or +Inf/-Inf/any-Inf, whether such improper (non)values come in from outside-world inputs or should emerge in own runtime-constructed `json.Value`s.

Now if this wasn't the stdlib but my own util-lib I'd wait until it becomes an issue in practice, then add to `.eql` an arg `useFloatEql: ?fn(f64, f64) bool` to have the user take care of the actual equality-desires their call-site situation imposes... but this smells a bit too ad-hoc/patchworky for a `json.Value`-owned method.

What sort of default handling would _you_ deem most sensible for a least-surprise general-purpose 'json.Value.eql(.Float, .Float)` semantics? My first gut feeling says "f1 == f2 or (math.isNan(f1) and math.isNan(f2)) or (math.isInf(f1) and math.isInf(f2))`... or, do so for Nan but ignore Inf? "Correctness" becomes so situational / fuzzy around the abyss-values of float types..  =/